### DISCUSSION TITLE
Be able to install a self-hosted out of Docker

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#22735] [Feature Requests]
### Summary

Fortunately I now use Docker and it rocks.

But I was curious to understand why `npm init directus-project test` doesn't work anymore as it should.
It systematically encounters a problem with sharp:
>npm init directus-project test
/var/www/test/node_modules/sharp/lib/sharp.js:114
  throw new Error(help.join('\n'));
        ^
>Error: Could not load the "sharp" module using the linux-x64 runtime
Possible solutions:
- Ensure optional dependencies can be installed:
    npm install --include=optional sharp
    yarn add sharp --ignore-engines
- Ensure your package manager supports multi-platform installation:
    See https://sharp.pixelplumbing.com/install#cross-platform
- Add platform-specific dependencies:
    npm install --os=linux --cpu=x64 sharp
- Consult the installation documentation:
    See https://sharp.pixelplumbing.com/install
    at Object.<anonymous> (/var/www/test/node_modules/sharp/lib/sharp.js:114:9)
    at Module._compile (node:internal/modules/cjs/loader:1356:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1414:10)
    at Module.load (node:internal/modules/cjs/loader:1197:32)
    at Module._load (node:internal/modules/cjs/loader:1013:12)
    at Module.require (node:internal/modules/cjs/loader:1225:19)
    at require (node:internal/modules/helpers:177:18)
    at Object.<anonymous> (/var/www/test/node_modules/sharp/lib/constructor.js:10:1)
    at Module._compile (node:internal/modules/cjs/loader:1356:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1414:10)
>Node.js v18.19.0
✖     Error while initializing the project
npm ERR! code 1
npm ERR! path /var/www
npm ERR! command failed
npm ERR! command sh -c cdp test

None of the proposed solutions work.

### Basic Example

run ` npm init directus-project test` on a bare new server on debian 12

### Motivation

Sometimes it's nice to have a quick test installation out of docker

### Detailed Design

Make sure sharp tries to install with the right options?

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

None

### Alternatives

Docker of course. But that is not the goal here.

### Adoption Strategy

Immediate

**Comment by br41nslug** [comment]
Directus is able to installed without Docker! But you may have to resolve some system specifc dependencies.

Directus does rely on a couple of system dependencies (`sharp`, `argon2` and `isolated-vm`) that may require you to manually set them up for your specific hardware/OS/Distro/configuration. Installing these when they fail to do so automatically can require some technical knowledge about the system you're administrating and reading up the dependencies documentation, `sharp` luckily gives you a lot of info about how to resolve this including links to get more information as you can see in the error.

Because it is not realistic for us to document all the various possible combinations (and more importantly keep those up-to-date) or provide support to every unique configuration we heavily recommend using Docker which solves this exact dependency problem.

**Comment by phazonoverload** [comment]
This will not be a supported path in the future - Docker is a standard tool which solves issues @br41nslug mentions. When we launch our upcoming docs platform, this install method will not feature.

**Reply by br41nslug** [reply]
It is already a path that's not actively supported or documented by us now 😄

**Comment by Pachat** [comment]
Thank you for your answers and your support.
I agree.

I now can do everything with Docker and it rocks :)
So let's go with this exclusive path in the future.

**Comment by fitu86** [comment]
It works for me in the following way

npm mkdir test
cd test
npm init
npm install directus
npx directus init

**Reply by rijkvanzanten** [reply]
It does for some, but not for everybody. It depends on your server environment and installed native dependencies. That's why we recommend using Docker to everybody, as that will work in every server environment that runs docker 👍🏻
